### DISCUSSION TITLE
Adding a location seems to not "stick"

### DISCUSSION THREAD

**Discussion by jas0420** [immich-app/immich#13829] [Q&A] [answered]
Hi all, I'm pretty new to Immich but loving it so far.  Having one difficulty that I was hoping someone could help me with.  I've added about 41,000 images to an external library over the past couple of weeks.  It is a combination of various generations of iPhone pics and various generations of Canon DSLR pics.  The latter doesn't have any native geolocation native to the images.

I'm on v1.119.1 currently and last night tried adding geo to several hundred of the ones that aren't tagged.  I was doing this by bulk-selecting them in the main "Photos" section, going to the menu in the top-right, selecting "Change Location",  scrolling around on the map, zooming in to where I wanted to be, droping the blue location marker, seeing the lat/lon fill in, and then clicking "Confirm".  Easy enough.

I spent about an hour doing this and then hopped over to the Map view to bask in the rewards of my efforts and...........  nothing had changed.  I still see plenty of iPhone images scattered across the map, but none of the ones that I had manually located within Immich showed. 

I tried refreshing the Map page, I tried hopping into DevTools and doing an "empty cache and hard reset" with no success.  

I then tried going into an individual picture and clicking the "Add a location" option and went through the same steps as above and didn't have any success that way either.

I'm doing an external library scan every 6 hours, so I held off reaching out here until a couple of those had run just in case there was something associated with that, but still no change unfortunately.

I have assumed that this would be adding the location to the database, not the EXIF of the image itself (so the fact that it's an External Library wouldn't matter).  Is that not accurate?

![immich-1](https://github.com/user-attachments/assets/8e80813b-9371-49a6-a0ff-6adfeeeac1bb)
![immich-2](https://github.com/user-attachments/assets/a8e7b982-15c2-48d3-a141-6b131a6ab56a)
![immich-3](https://github.com/user-attachments/assets/130896f0-9853-4c77-bbb2-003852f0d15c)
![immich-1](https://github.com/user-attachments/assets/83954adf-3721-428d-a485-f4afbefaf508)

**Comment by bo0tzz** [comment]
Is your external library mounted readonly?

**Reply by jas0420** [reply]
it is, yes.

**Reply by bo0tzz** [reply, marked as answer]
There's the issue. Metadata edits are written to an XMP sidecar and then read back, which fails on a readonly library.

**Reply by jas0420** [reply]
Ah, that makes sense then.  I had originally set it up as RO as a safety net, but I think this new path where I've pointed the external library too can just become an Immich-specific copy of the source files (which are in a mess of an organizational condition).  If I do that, I don't have as much hesitation about switching to RW on the mount..........  I'll give that a try.  Thx @bo0tzz !

**Comment by jas0420** [comment]
Just circling back to confirm that removing the RO criteria from the photos mount in the compose.yaml did indeed solve this for me.  thank you again!
